### PR TITLE
Fix binary build cmake flags forwarding

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: docker/packager/binary/build.sh [hunk 1]
```diff
@@ -17,7 +17,9 @@ ccache --show-stats ||:
 ccache --zero-stats ||:
 ln -s /usr/lib/x86_64-linux-gnu/libOpenCL.so.1.0.0 /usr/lib/libOpenCL.so ||:
 rm -f CMakeCache.txt
-cmake --debug-trycompile --verbose=1 -DCMAKE_VERBOSE_MAKEFILE=1 -LA "-DCMAKE_BUILD_TYPE=$BUILD_TYPE" "-DSANITIZE=$SANITIZER" -DENABLE_CHECK_HEAVY_BUILDS=1 "$CMAKE_FLAGS" ..
+
+read -ra CMAKE_FLAGS <<< "${CMAKE_FLAGS:-}"
```
Comment (alexey-milovidov) [inline docker/packager/binary/build.sh:21]: Maybe write a comment? (Does it mean "read into array"?)
Or this trash script does not deserve a comment?

**Review by alexey-milovidov — APPROVED** [review]

File: docker/packager/binary/build.sh (hunk 1 above)
Comment (alesapin) [reply docker/packager/binary/build.sh:21]: I've never seen this syntax before but seems like bash spellchecker requires to use array syntax for flags...

**Comment by alesapin** [comment]
For some reason, `clang-10-aarch64` broken now....

File: CMakeLists.txt [hunk 2]
```diff
@@ -514,7 +514,8 @@ macro (add_executable target)
     # invoke built-in add_executable
     # explicitly acquire and interpose malloc symbols by clickhouse_malloc
     # if GLIBC_COMPATIBILITY is ON and not sanitizer build, provide memcpy symbol explicitly to neutrialize thinlto's libcall generation.
-    if (GLIBC_COMPATIBILITY AND NOT SANITIZE)
+    # `flto=thin` used only for release build, probably this `if` should be more strict.
+    if (GLIBC_COMPATIBILITY AND NOT SANITIZE AND NOT ARCH_AMD64)
```
Comment (danlark1) [inline CMakeLists.txt:518]: why ARCH_AMD64?

Comment (alesapin) [reply CMakeLists.txt:518]: Oh thanks, incorrect flag. I was trying to fix https://clickhouse-builds.s3.yandex.net/15595/b3fa86191e4f85ad3ff4d72d6370d56b80be868c/clickhouse_special_build_check/build_log_788464832_1601815120.txt

**Comment by alesapin** [comment]
Build OK, Compatibility OK, ThinLTO ok.

**Pull request merged**